### ISSUE TITLE
@Hidden does not work for fields named in upper camel case style

### ISSUE THREAD

**Issue by Strunevskiy** [swagger-api/swagger-core#3560]
When there are fields named in upper camel case style using ```@JsonProperty``` annotation that is marked by   Hidden annotation , fields are displayed on swagger-ui

```java
public class Foo {

	@Schema(name = "FirstName", hidden = true)
	@JsonProperty(value = "FirstName")
	@Hidden
	private String firstName;

	@JsonProperty(value = "lastName")
	private String lastName;

	public String getFirstName() {
		return firstName;
	}
	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}
}
```

```java
ResolvedSchema resolvedSchema = ModelConverters.getInstance()
		.resolveAsResolvedSchema(new AnnotatedType(Foo.class));
if (resolvedSchema.schema != null) {
	Schema schemaN = resolvedSchema.schema;
	Map<String, Schema> schemaMap = resolvedSchema.referencedSchemas;
	Assert.assertTrue( schemaMap.get("Foo").getProperties().size()==1, "Foo should only contain lastName");

}
```

**Comment by rgroothuijsen** [comment]
This appears to actually be the result of `@JsonProperty` rather than any particular naming conventions. When I add `@Hidden` to the getter and setter as well, the field disappears again.

**Comment by Strunevskiy** [comment]
@rgroothuijsen  do I understand you correctly if I add @Hidden to the setter and getter the field will disappear ?

**Comment by rgroothuijsen** [comment]
Yes, that's what I found.

**Comment by frantuma** [comment]
Thanks for reporting this. should be fixed in #3706, please see comment there about behaviour change.
